clippy_lints/src/matches/match_like_matches.rs

```diff
@@ -170,3 +180,16 @@ fn find_bool_lit(ex: &ExprKind<'_>) -> Option<bool> {
         _ => None,
     }
 }
+
+fn is_some(path_kind: PatKind<'_>) -> bool {
+    match path_kind {
+        PatKind::TupleStruct(QPath::Resolved(_, path), patterns, _) if is_wild(&patterns[0]) => {
```

Comment:
Optional suggestion: Here you can destruct the patterns slice directly, just a small quality of life thing and to avoid panics from indexing, even though this should be 100 save
```suggestion
        PatKind::TupleStruct(QPath::Resolved(_, path), [first, ..], _) if is_wild(first) => {
```